torchtext/data/metrics.py

```diff
@@ -0,0 +1,93 @@
+import math
+import collections
+from torchtext.data.utils import ngrams_iterator
+
+
+def _compute_ngram_counter(tokens, max_n):
+    """ Create a Counter with a count of unique n-grams in the tokens list
+
+    Arguments:
+        tokens: a list of tokens (typically a string split on whitespaces)
+        max_n: the maximum order of n-gram wanted
+
+    Outputs:
+        output: a collections.Counter object with the unique n-grams and their
+            associated count
+
+    Examples:
+        >>> from torchtext.data.metrics import _compute_ngram_counter
+        >>> tokens = ['me', 'me', 'you']
+        >>> _compute_ngram_counter(tokens, 2)
```

Comment:
Do you want people to call `_compute_ngram_counter` directly? If so, you should keep the func name publicly.

Reply:
The idea is that this function can be shared amongst all the other metrics (ROUGE, etc.), since they are all very similar in nature. Once those are implemented and depending on the added complexity we might end up deciding that this function either needs to be public or isn't necessary (since it's indeed only one line).